client/service/src/lib.rs

```diff
@@ -323,11 +323,18 @@ where
 		}
 	}
 
+	let random_port = |mut addr: SocketAddr| {
+		addr.set_port(0);
+		addr
+	};
+
 	let ws_addr = config.rpc_ws.unwrap_or_else(|| "127.0.0.1:9944".parse().unwrap());
+	let ws_addr2 = random_port(ws_addr);
 	let http_addr = config.rpc_http.unwrap_or_else(|| "127.0.0.1:9933".parse().unwrap());
```

Comment:
Not a part of this PR, but we should probably except here with a `qed`, unless the rules changed?